File: 3334-find-the-maximum-factor-score-of-array/3334-find-the-maximum-factor-score-of-array.cpp
class Solution {
public:
    long long maxScore(vector<int>& nums) {
        auto factor = [&](const vector<int>& v) -> long long {
            if(v.empty()) return 0LL;
            long long g = v[0];
            long long l = v[0];
            for(int i=1;i<v.size();i++){
                g = gcd(g, (long long)v[i]);
                l = l / gcd(l, (long long)v[i]) * v[i];
            }
            return g * l;
        };
        long long m = factor(nums);
        for(int i=0;i<nums.size();i++){
            vector<int> tmp;
            for(int j=0;j<nums.size();j++) if(j!=i) tmp.push_back(nums[j]);
            m = max(m, factor(tmp));
        }
        return m;
    }
};

File: 3334-find-the-maximum-factor-score-of-array/3334-find-the-maximum-factor-score-of-array.java
import java.math.BigInteger;
class Solution {
    public long maxScore(int[] nums) {
        long max = 0;
        for (int i = -1; i < nums.length; i++) {
            BigInteger gcd = BigInteger.ZERO, lcm = BigInteger.ONE;
            for (int j = 0; j < nums.length; j++) {
                gcd = i == j ? gcd : gcd.gcd(BigInteger.valueOf(nums[j]));
                lcm = i == j ? lcm : lcm.multiply(BigInteger.valueOf(nums[j])).divide(lcm.gcd(BigInteger.valueOf(nums[j])));
            }
            max = Math.max(max, gcd.multiply(lcm).longValue());
        }
        return max;
    }
}